services/business/istio_validations.go

```diff
@@ -42,6 +44,41 @@ func (in *IstioValidationsService) GetServiceValidations(namespace, service stri
 	return runObjectCheckers(objectCheckers), nil
 }
 
+// GetDeploymentsValidations returns an IstioValidations object with all the checks found when running
+// all the enabled checkers.
+func (in *IstioValidationsService) GetDeploymentsValidations(namespace string) (models.IstioValidations, error) {
+	// Get all the Istio objects from a Namespace and deployment
+	istioDetails, err := in.k8s.GetDeployments(namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	objectCheckers := []ObjectChecker{
+		checkers.DeploymentChecker{Deployments: istioDetails},
+	}
+
+	// Get groupal validations for same kind istio objects
+	return runObjectCheckers(objectCheckers), nil
+}
+
+// GetDeploymentValidations returns an IstioValidations object with all the checks found when running
+// all the enabled checkers.
+func (in *IstioValidationsService) GetDeploymentValidations(namespace, deployment string) (models.IstioValidations, error) {
+	// Get all the Istio objects from a Namespace and deployment
+	istioDetails, err := in.k8s.GetDeployment(namespace, deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	deploymentList := v1beta1.DeploymentList{Items: []v1beta1.Deployment{*istioDetails}}
+	objectCheckers := []ObjectChecker{
+		checkers.DeploymentChecker{Deployments: &deploymentList},
```

Comment:
I guess that here we should add the pod sidecar presence checker to validate each pod of the deployment.
We have something similar for the [service details validations](https://github.com/kiali/kiali/blob/4f8168c1ec077acaf16be89b7977bd2a33bd9ece/services/business/istio_validations.go#L35).